### ISSUE TITLE
[Question] Can we train and forecast given incomplete `input_chunk_length` history ?

### ISSUE THREAD

**Issue by fred-navruzov** [unit8co/darts#1474]
Hi guys!
First of all, thank you for your fantastic work on unifying time series forecasting!

I got stuck - after browsing examples and documentation, I wonder whether there exists an explicit (or implicit) way to forecast `output_chunk_length` given **incomplete** historical data, to deal with cold-start problems?
It might be beneficial for **both train and inference for many industries** to have such functionality covered

Consider the following example, let us have: 
- `input_chunk_length` = 20 (*maximum* historical window to train on)
- `output_chunk_length` = 10 (desired forecast horizon)
- we have static_covariates, past_covariates (of *maximum* length 10), future covariates (past and future)
- 2 timeseries of **length 15** (so it's less than `input_chunk_length` + `output_chunk_length` = 30 but it's **strictly longer** than `output_chunk_length`)

1) how can we **train** model by sampling chunks of [19 masked rows, 1st historical row] to predict [2, 3, ..., 11] timestamps of this timeseries, ..., up to [15 masked rows , 1st historical row, ..., 5th historical row]  to predict [6, 7, ..., 15] rows of this timeseries? 

2) after trained model **learn how to predict each timestep of timeseries starting from 2nd and beyond** (given 1st and some masked rows as an input, so we can deal with  cold start), how can we **predict** from *such incomplete historical* data?

Right now, I can't imagine any other way to do it with `darts` framework, rather than add `fake start` to timeseries consisting of (`input_chunk_length` - 1) fake target rows, the same `fake start` addition for `past_covariates` data
The bad thing is, I can't explicitly use `NaN`s here for neural-network based models, so that's not masking and that will affect the loss and, as a result, bias the model

Any suggestions / future development plans that include such workaround will be greatly appreciated!

**Comment by hrzn** [comment]
Hi @fred-navruzov and thanks for raising this topic. I think we could imagine something for a subset of the models (eg RNNs, sequence models), for which masking can be used to instruct the model that some time stamps are missing and should be ignored. For many other models (tabular, or "block" neural nets), I don't think there's a good solution, as the models expect vectors of fixed size in input, and so the solution would be to "pad" the inputs with some filling values. This would be equivalent to filling missing values on the original input series (with e.g. a 0). But it's not perfect either as it can create a bias.

**Comment by fred-navruzov** [comment]
> Hi @fred-navruzov and thanks for raising this topic. I think we could imagine something for a subset of the models (eg RNNs, sequence models), for which masking can be used to instruct the model that some time stamps are missing and should be ignored. For many other models (tabular, or "block" neural nets), I don't think there's a good solution, as the models expect vectors of fixed size in input, and so the solution would be to "pad" the inputs with some filling values. This would be equivalent to filling missing values on the original input series (with e.g. a 0). But it's not perfect either as it can create a bias.

@hrzn, thanks for your feedback!
Yes, I was particularly interested in seq2seq models (i.e. TFTModel or DLinear) that can (potentially) use masking for that purpose.
Do you guys have plans to add abovementioned support for a subset of models to your amazing framework in nearest releases? In case you will, I can't help you with code production itself but I can raise another `enchancement` ticket with some examples and use cases for you to not forget about it

**Comment by hrzn** [comment]
Thanks @fred-navruzov. For now there is no effort in this direction, but I think it'd be good to add something to our backlog, so we can track it and evaluate potential approaches. So an improvement proposal with some examples would be welcome yes. I unfortunately cannot tell you if/when we will tackle this - it'll be best effort :)
